scripts/prepare_network.py

```diff
@@ -90,86 +91,48 @@ def add_emission_prices(n, emission_prices=None, exclude_co2=False):
-        n.lines.type.map(n.line_types.i_nom) *
-        n.lines.bus0.map(n.buses.v_nom)
-    )
-
-    n.lines['capital_cost_lc'] = n.lines['capital_cost']
-    n.links['capital_cost_lc'] = n.links['capital_cost']
-    total_line_cost = ((lines_s_nom * n.lines['capital_cost_lc']).sum() +
-                       n.links.loc[links_dc_b].eval('p_nom * capital_cost_lc').sum())
-
-    if lc == 'opt':
-        costs = load_costs(Nyears, snakemake.input.tech_costs,
-                           snakemake.config['costs'], snakemake.config['electricity'])
-        update_transmission_costs(n, costs, simple_hvdc_costs=False)
-    else:
-        # Either line_volume cap or cost
-        n.lines['capital_cost'] = 0.
-        n.links.loc[links_dc_b, 'capital_cost'] = 0.
 
-    if lc == 'opt' or float(lc) > 1.0:
-        n.lines['s_nom_min'] = lines_s_nom
-        n.lines['s_nom_extendable'] = True
-
-        n.links.loc[links_dc_b, 'p_nom_min'] = n.links.loc[links_dc_b, 'p_nom']
-        n.links.loc[links_dc_b, 'p_nom_extendable'] = True
-
-        if lc != 'opt':
-            line_cost = float(lc) * total_line_cost
-            n.add('GlobalConstraint', 'lc_limit',
-                  type='transmission_expansion_cost_limit',
-                  sense='<=', constant=line_cost, carrier_attribute='AC, DC')
-    return n
-
-def set_line_volume_limit(n, lv, Nyears=1.):
+def set_transmission_limit(n, ll_type, factor, Nyears=1):
     links_dc_b = n.links.carrier == 'DC' if not n.links.empty else pd.Series()
 
-    lines_s_nom = n.lines.s_nom.where(
-        n.lines.type == '',
-        np.sqrt(3) * n.lines.num_parallel *
-        n.lines.type.map(n.line_types.i_nom) *
-        n.lines.bus0.map(n.buses.v_nom)
-    )
+    _lines_s_nom = (np.sqrt(3) * n.lines.type.map(n.line_types.i_nom) *
+                   n.lines.num_parallel *  n.lines.bus0.map(n.buses.v_nom))
+    lines_s_nom = n.lines.s_nom.where(n.lines.type == '', _lines_s_nom)
 
-    total_line_volume = ((lines_s_nom * n.lines['length']).sum() +
-                         n.links.loc[links_dc_b].eval('p_nom * length').sum())
 
-    if lv == 'opt':
-        costs = load_costs(Nyears, snakemake.input.tech_costs,
-                           snakemake.config['costs'], snakemake.config['electricity'])
-        update_transmission_costs(n, costs, simple_hvdc_costs=True)
-    else:
-        # Either line_volume cap or cost
-        n.lines['capital_cost'] = 0.
-        n.links.loc[links_dc_b, 'capital_cost'] = 0.
+    col = 'capital_cost' if ll_type == 'c' else 'length'
+    ref = (lines_s_nom @ n.lines[col] +
+           n.links[links_dc_b].p_nom @ n.links[links_dc_b][col])
+
+    costs = load_costs(Nyears, snakemake.input.tech_costs,
+                       snakemake.config['costs'],
+                       snakemake.config['electricity'])
+    update_transmission_costs(n, costs, simple_hvdc_costs=False)
 
-    if lv == 'opt' or float(lv) > 1.0:
+    if factor == 'opt' or float(factor) > 1.0:
```

Comment:
Wondering whether to add an assert statement or warning if factor is lower than 1. In this case transmission network is not extendable and no constraints are added.

Reply:
The code now support lv/lc limits below 1. Then no lower capacity limits are set but the constraint is still set.